### DISCUSSION TITLE
Whether `local_ts` must be configured in remote access mode

### DISCUSSION THREAD

**Discussion by zzzzrf** [strongswan/strongswan#2126] [Q&A] [answered]
hello~

I noticed while learning the [remote access mode](https://www.strongswan.org/testing/testresults/ikev2/rw-cert/) that the central point device will be configured with a stream of interest`local_ts = 10.1.0.0/24`. Roadwarriors will also be configured with a stream of interset`remote_ts = 10.1.0.0/16`

When negotiating IPSec, strongswan seems to choose the intersection of these two streams of interest:`10.1.0.0/24`。

```log
14[IKE] CHILD_SA net{1} established with SPIs ceae680e_i c835409c_o and TS 10.1.0.0/24 === 10.3.0.1/32
```

**My question is**：
Is there a configuration so that the central point device does not need to configure `local_ts` and can automatically select the flow of interest based on the request of the roadwarriors?

For example：

* The central point device

```conf
connections{
    rw {
        local_addrs = 10.10.100.174
        local {
            auth = psk
            id = server
        }
        remote {
            auth = psk
        }
        children {
            net {
                # The center point does not configure `local_ts`
                # local_ts = 192.168.1.0/24
                esp_proposals = aes128-sha1
            }
        }
        version = 1
        proposals = aes128-sha1-modp1024
        mobike = no
    }
}
```

* roadwarriors

```conf
connections {
    home {
        local_addrs  = 10.10.100.173
        remote_addrs = 10.10.100.174

    local {
        auth = psk
        id = client
    }
    remote {
        auth = psk
        id = server
    }
    children {
        net {
            esp_proposals = aes128-sha1
            remote_ts = 192.168.1.0/24
        }
    }
        version = 1
        proposals = sha1-aes128-modp1024
        mobike = no
    }
}
```

I tried this configuration and the negotiation failed.

```log
06[IKE] no matching CHILD_SA config found for 10.10.100.173/32 === 192.168.1.0/24
06[ENC] generating INFORMATIONAL_V1 request 4253650877 [ HASH N(INVAL_ID) ]
```

**Comment by tobiasbrunner** [comment, marked as answer]
> Is there a configuration so that the central point device does not need to configure `local_ts` and can automatically select the flow of interest based on the request of the roadwarriors?

Yes, traffic selector narrowing allows that. Just configure `local_ts = 0.0.0.0/0` then a client can propose whatever it likes to send to the server via `remote_ts` (could be a single subnet, or multiple). The negotiated traffic selector will be narrowed to that.

But note that this really allows the clients to send anything they like, they could propose `remote_ts = 0.0.0.0/0` to tunnel all traffic to the server. Of course, whether that's a problem depends on your use case and the firewall config on the server.



> I tried this configuration and the negotiation failed.

That's because if you don't configure `local_ts` it will default to the local IP address (as you can see in the log).

**Reply by zzzzrf** [reply]
Thank you very much ~
